commons/src/test/java/com/ge/predix/acs/commons/web/BaseRestApiControllerAdviceTest.java

```diff
@@ -19,38 +19,134 @@
+                .handleHttpMediaTypeNotAcceptableException(e);
+
+        Assert.assertNotNull(actualErrorResponse);
+        Assert.assertTrue(actualErrorResponse.getErrorDetails().getErrorCode().equals("FAILED"));
+        Assert.assertTrue(actualErrorResponse.getErrorDetails().getErrorMessage().equals("Not Acceptable"));
+    }
+
+    @Test
+    public void testBaseRestApiControllerAdviceHttpRequestMethodNotSupportedException() {
+
+        BaseRestApiControllerAdvice baseRestApiControllerAdvice = new BaseRestApiControllerAdvice();
+
+        HttpRequestMethodNotSupportedException e = new HttpRequestMethodNotSupportedException("GET");
+
+        RestApiErrorResponse actualErrorResponse = baseRestApiControllerAdvice
+                .handleHttpRequestMethodNotSupportedException(e);
+
+        Assert.assertNotNull(actualErrorResponse);
+        Assert.assertTrue(actualErrorResponse.getErrorDetails().getErrorCode().equals("FAILED"));
+        Assert.assertTrue(actualErrorResponse.getErrorDetails().getErrorMessage()
+                .equals("Method Not Allowed"));
+    }
+
+    @Test
+    public void testBaseRestApiControllerAdviceIllegalArgumentException() {
+
+        BaseRestApiControllerAdvice baseRestApiControllerAdvice = new BaseRestApiControllerAdvice();
+
+        IllegalArgumentException e = new IllegalArgumentException();
+
+        RestApiErrorResponse actualErrorResponse = baseRestApiControllerAdvice.handleIllegalArgumentException(e);
+
+        Assert.assertNotNull(actualErrorResponse);
+        Assert.assertTrue(actualErrorResponse.getErrorDetails().getErrorCode().equals("FAILED"));
+    }
+
+    @Test
+    public void testBaseRestApiControllerAdviceUntrustedIssuerException() {
+
+        BaseRestApiControllerAdvice baseRestApiControllerAdvice = new BaseRestApiControllerAdvice();
+
+        UntrustedIssuerException e = new UntrustedIssuerException();
+
+        RestApiErrorResponse actualErrorResponse = baseRestApiControllerAdvice
+                .handleUntrustedIssuerOrSecurityException(e);
 
-        Mockito.when(errorHandler.createApiErrorResponse(e, request, response)).thenReturn(errorResponse);
+        Assert.assertNotNull(actualErrorResponse);
+        Assert.assertTrue(actualErrorResponse.getErrorDetails().getErrorCode().equals("FAILED"));
+
+    }
+
+    @Test
+    public void testBaseRestApiControllerAdviceHttpMessageNotReadableException() {
 
         BaseRestApiControllerAdvice baseRestApiControllerAdvice = new BaseRestApiControllerAdvice();
-        baseRestApiControllerAdvice.setRestErrorHandler(errorHandler);
 
-        ModelAndView actualErrorResponse = baseRestApiControllerAdvice.handleException(e, request, response);
+        HttpServletRequest request = new MockHttpServletRequest();
+        HttpServletResponse response = new MockHttpServletResponse();
+        HttpMessageNotReadableException e = new HttpMessageNotReadableException("Malformed JSON syntax.");
```

Comment:
Initialize exception with a different custom message and then assert that response error message contains "Malformed JSON syntax"  in addition to your custom message.